deploy.sh

```diff
@@ -9,16 +9,19 @@ DEPLOY_PATH=./dist
 
 DEPLOY_SUBDOMAIN_UNFORMATTED_LIST=()
 
-if [ "$NODE_ENV" == "test" ]
-then
-  echo 'No need to deploy build for UI tests, exiting'
-  exit 0
-fi
-
 if [ "$TRAVIS_PULL_REQUEST" != "false" ]
 then
-  DEPLOY_SUBDOMAIN_UNFORMATTED_LIST+=(${TRAVIS_PULL_REQUEST}-pr)
-elif [ -n "${TRAVIS_TAG// }" ] #TAG is not empty
+  if [ "$NODE_ENV" == "production" ]
+  then
+    DEPLOY_SUBDOMAIN_UNFORMATTED_LIST+=(release-${TRAVIS_PULL_REQUEST}-pr)
+  elif [ "$NODE_ENV" == "staging" ]
+  then
+    DEPLOY_SUBDOMAIN_UNFORMATTED_LIST+=(staging-${TRAVIS_PULL_REQUEST}-pr)
+  elif [ "$NODE_ENV" == "test" ]
+  then
+    DEPLOY_SUBDOMAIN_UNFORMATTED_LIST+=(${TRAVIS_PULL_REQUEST}-pr)
+  fi
+elif [ -n "${TRAVIS_TAG// }" ]  && [ "$NODE_ENV" == "production" ] #TAG is not empty
```

Comment:
the production clause should be evaluated inside of this if, otherwise the else below (for branches) can evaluate to true even it's a tag change.

on the branch clause we probably only want to evaluate in case it's `test `